scripts/paragraph_segmentation_dcnn_pretraining.py

```diff
@@ -0,0 +1,160 @@
+
+batch_size = 32
+
+def transform(data, label):
+    image = mx.nd.array(data).expand_dims(axis=2)
+    image = resize_short(image, int(800/3))
+    image = image.transpose([2, 0, 1])/255.
+    label = label[0].astype(np.float32)
+    return image, mx.nd.array(label)
+
+def augment_transform(data, label):
+    width_translation_range = 0.05
+    height_translation_range = 0.05
+
+    ty = random.uniform(-height_translation_range, height_translation_range)
+    tx = random.uniform(-width_translation_range, width_translation_range)
+    st = tf.SimilarityTransform(translation=(tx*data.shape[1], ty*data.shape[0]))
+    data = tf.warp(data, st)
+
+    label[0][0] = label[0][0] - tx
+    label[0][1] = label[0][1] - ty
+
+    return transform(data*255., label)
+    
+train_data = gluon.data.DataLoader(train_ds.transform(augment_transform), batch_size, shuffle=True, num_workers=multiprocessing.cpu_count())
+test_data = gluon.data.DataLoader(test_ds.transform(transform), batch_size, shuffle=False, num_workers=multiprocessing.cpu_count())
+    
+def make_cnn():
+    p_dropout = 0.5
+
+    cnn = gluon.nn.HybridSequential()
+    cnn.add(gluon.nn.Conv2D(kernel_size=(3,3), padding=(1,1), channels=16, activation="relu"))
+    cnn.add(gluon.nn.BatchNorm())
+
+    cnn.add(gluon.nn.Conv2D(kernel_size=(3,3), padding=(1,1), channels=16, activation="relu"))
+    cnn.add(gluon.nn.BatchNorm())
+
+    cnn.add(gluon.nn.Conv2D(kernel_size=(3,3), padding=(1,1), channels=16, activation="relu"))
+    cnn.add(gluon.nn.BatchNorm())
+
+    cnn.add(gluon.nn.Conv2D(kernel_size=(3,3), padding=(1,1), channels=16, activation="relu"))
+    cnn.add(gluon.nn.MaxPool2D(pool_size=(2,2), strides=(2,2)))
+    cnn.add(gluon.nn.BatchNorm())
+
+    cnn.add(gluon.nn.Conv2D(kernel_size=(3,3), padding=(1,1), channels=16, activation="relu"))
+    cnn.add(gluon.nn.MaxPool2D(pool_size=(2,2), strides=(2,2)))
+    cnn.add(gluon.nn.BatchNorm())
+
+    cnn.add(gluon.nn.Flatten())
+    cnn.add(gluon.nn.Dense(64, activation='relu'))
+    cnn.add(gluon.nn.Dropout(p_dropout))
+    cnn.add(gluon.nn.Dense(64, activation='relu'))
+    cnn.add(gluon.nn.Dropout(p_dropout))
+    cnn.add(gluon.nn.Dense(4, activation='sigmoid'))
+
+    cnn.hybridize()
+    cnn.collect_params().initialize(mx.init.Normal(), ctx=ctx)
+    return cnn
+
+cnn = make_cnn()
+l2_loss = gluon.loss.L2Loss()
+LEARNING_RATE = 0.001
+trainer = gluon.Trainer(cnn.collect_params(), 'adam', {'learning_rate': LEARNING_RATE, })
+
+epochs = 500
+print_every_n = 20
+send_image_every_n = 20
+save_every_n = 100
+
+for e in range(epochs):    
+    loss = nd.zeros(1, ctx)
+    test_loss = nd.zeros(1, ctx)
+    tick = time.time()
+    acc = nd.zeros(1, ctx)
+    for i, (data, label) in enumerate(train_data):
+        data = data.as_in_context(ctx)
+        label = label.as_in_context(ctx)
+        
+        with autograd.record():
+            output = cnn(data)
+            loss_l2 = l2_loss(output, label)
```

Comment:
is this the only difference between the IOU loss and the pretraining script?
Please combine and use a parameter to switch the loss, and another one to point to a give checkedpoint model